File: Data Structures & Algorithms/permutations/submission-3.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        
        res = [[]]

        for num in nums:
            new_perms = []
            for r in res:
                for i in range(len(r)+1):
                    p_copy = r.copy()
                    p_copy.insert(i, num)
                    new_perms.append(p_copy)
            
            res = new_perms
        
        return res

File: Data Structures & Algorithms/maximum-product-subarray/submission-3.java
class Solution {
    public int maxProduct(int[] nums) {
        int n = nums.length;
        int res = Integer.MIN_VALUE;

        for(int i=0; i<n; i++){
            int curr = 1;
            for(int j=i; j<n; j++){
                curr *= nums[j];
                if(curr > res){
                    res = curr;
                }
            }
        }

        return res;
    }
}
